src/request_queues.js

```diff
@@ -0,0 +1,340 @@
+
+        checkParamOrThrow(baseUrl, 'baseUrl', 'String');
+        checkParamOrThrow(token, 'token', 'String');
+        checkParamOrThrow(limit, 'limit', 'Maybe Number');
+        checkParamOrThrow(offset, 'offset', 'Maybe Number');
+        checkParamOrThrow(desc, 'desc', 'Maybe Boolean');
+        checkParamOrThrow(unnamed, 'unnamed', 'Maybe Boolean');
+
+        const query = { token };
+
+        if (limit) query.limit = limit;
+        if (offset) query.offset = offset;
+        if (desc) query.desc = 1;
+        if (unnamed) query.unnamed = 1;
+
+        return requestPromise({
+            url: `${baseUrl}${BASE_PATH}`,
+            json: true,
+            method: 'GET',
+            qs: query,
+        })
+            .then(pluckData);
+    },
+
+    /**
+     * Gets request queue.
+     *
+     * @memberof ApifyClient.requestQueues
+     * @instance
+     * @param {Object} options
+     * @param {String} options.queueId - Unique queue ID
+     * @param callback
+     * @returns {RequestQueue}
+     */
+    getQueue: (requestPromise, options) => {
+        const { baseUrl, queueId } = options;
+
+        checkParamOrThrow(baseUrl, 'baseUrl', 'String');
+        checkParamOrThrow(queueId, 'queueId', 'String');
+
+        return requestPromise({
+            url: `${baseUrl}${BASE_PATH}/${queueId}`,
+            json: true,
+            method: 'GET',
+        })
+            .then(pluckData)
+            .catch(catchNotFoundOrThrow);
+    },
+
+    /**
+     * Deletes request queue.
+     *
+     * @memberof ApifyClient.requestQueues
+     * @instance
+     * @param {Object} options
+     * @param {String} options.queueId - Unique queue ID
+     * @param callback
+     * @returns {*}
+     */
+    deleteQueue: (requestPromise, options) => {
+        const { baseUrl, queueId } = options;
+
+        checkParamOrThrow(baseUrl, 'baseUrl', 'String');
+        checkParamOrThrow(queueId, 'queueId', 'String');
+
+        return requestPromise({
+            url: `${baseUrl}${BASE_PATH}/${queueId}`,
+            json: true,
+            method: 'DELETE',
+        });
+    },
+
+    /**
+     * @typedef {Object} AddedRequetInfo
+     * @property {Boolean} wasAlreadyPresent Indicates if request was already present in the queue.
+     * @property {Boolean} wasAlreadyHandled Indicates if request was already marked as handled.
+     * @property {String} id The ID of the added request
+     */
+
+    /**
+     * Adds request to queue.
+     *
+     * @memberof ApifyClient.requestQueues
+     * @instance
+     * @param {Object} options
+     * @param {String} options.queueId - Unique queue ID
+     * @param {Object} options.request - Request object
+     * @param {Object} [options.putInFront] - If yes then request will be enqueued to the begining of the queue
+     *                                        and to the end of the queue otherwise.
+     * @param callback
+     * @returns {AddedRequetInfo}
```

Comment:
Returns a Promise that resolves to AddedRequestInfo (there is also a typo in AddedRequetInfo)